langlib/langlib-test/src/test/resources/test-src/valuelib_test.bal

```diff
@@ -517,6 +517,30 @@ type Person2 record {
     int age;
 };
 
+function testCloneWithTypeTupleToJSON() {
+    [string, string, string] tupleValue1 = ["Mohan", "single", "LK2014"];
```

Comment:
Please add tests with types like
- [string, xml|int]
- [string, anydata...]
- [string, int|xml...]

Both positive and negative.